### PR TITLE
Highcharts

### PR DESCRIPTION
**Pull request by mayabrandi**
I realized we could not continue using charts for the plotting. It was not complex enough to generate the plots we would need. Sfter some investigations we decided on using Highcharts instead. This PR is the switch from charts to highcharts. It includes changes in:

utils.py - to restructure the input that is given to the plots
views.py - to handle the updates of the utils.py functions
XX.html - Most htmls are changed to produce highcharts plots instead of charts. 

All code has been tested locally

### CODE REVIEW

File: vogue/server/utils.py [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
-         ('RGB(23, 32, 42)','RGB(23, 32, 42, 0.2)'),('RGB(183, 149, 11)','RGB(183, 149, 11, 0.2)'),  
-        ('RGB(0, 255, 0)','RGB(0, 255, 0, 0.2)'),('RGB(0, 255, 255)','RGB(0, 255, 255, 0.2)'),
-        ('RGB(255, 0, 255)','RGB(255, 0, 255, 0.2)'),('RGB(0, 0, 255)','RGB(0, 0, 255, 0.2)')]
+          ('RGB(0, 128, 128)','RGB(0, 128, 128, 0.2)'),('RGB(128, 0, 128)','RGB(128, 0, 128, 0.2)'),
+          ('RGB(128, 0, 0)','RGB(128, 0, 0,0.2)'), ('RGB(128, 128, 0)','RGB(128, 128, 0,0.2)'),
+          ('RGB(52, 152, 219)', 'RGB(52, 152, 219, 0.2)'),('RGB(33, 97, 140)','RGB(33, 97, 140, 0.2)'),  
+          ('RGB(46, 204, 113)','RGB(46, 204, 113, 0.2)'),('RGB(241, 196, 15)','RGB(241, 196, 15, 0.2)'),
+          ('RGB(23, 32, 42)','RGB(23, 32, 42, 0.2)'),('RGB(183, 149, 11)','RGB(183, 149, 11, 0.2)'),  
+          ('RGB(0, 255, 0)','RGB(0, 255, 0, 0.2)'),('RGB(0, 255, 255)','RGB(0, 255, 255, 0.2)'),
+          ('RGB(255, 0, 255)','RGB(255, 0, 255, 0.2)'),('RGB(0, 0, 255)','RGB(0, 0, 255, 0.9)')]
```
Comment (hassanfa) [inline vogue/server/utils.py]: Is it possible to move these to `vogue/models`?

File: vogue/server/utils.py [hunk 2]
```diff
@@ -56,78 +58,85 @@ def get_average(samples: list, key: str)-> float:
...
-        group_by: the data in the plot will be grouped by the the values in this list.
-        group_key: is the key in the database holding the group value."""
+    values = []
+    percentiles = []
+    for sample in samples:
+        value = sample.get(key)
+        if isinstance(value, int) or isinstance(value, float):
+            values.append(value)
+    if values:
+        a = np.array(values)
```
Comment (hassanfa) [inline vogue/server/utils.py]: I know `a` is an intermediate variable, but is it possible to rename it to something more descriptive? (I'm also guilty of doing the same :-)

File: vogue/server/utils.py [hunk 3]
```diff
@@ -56,78 +58,85 @@ def get_average(samples: list, key: str)-> float:
...
+        percentiles = [np.percentile(a,25), np.percentile(a,50), np.percentile(a,72)]
```
Comment (hassanfa) [inline vogue/server/utils.py]: Do you mean `np.percentile(a,75)`? You've written `72`?

File: vogue/server/templates/navigation.html [hunk 4]
```diff
@@ -1,43 +1,50 @@
...
+      </li>
+      <li class="nav-item dropdown">
+        <a class="nav-link dropdown-toggle" href="#" id="navbarDropdown" role="button" data-toggle="dropdown"
+          aria-haspopup="true" aria-expanded="false">
+          Common trends
+        </a>
+        <div class="dropdown-menu" aria-labelledby="navbarDropdown">
+          <a class="dropdown-item" href="{{ url_for('turn_around_times', year_of_interest=year_of_interest) }}">Turnaround
+            Times</a>
+          <a class="dropdown-item" href="{{ url_for('comon_samples', year_of_interest=year_of_interest) }}">Samples</a>
```
Comment (hassanfa) [inline vogue/server/templates/navigation.html]: typo: common_samples

File: vogue/server/views.py [hunk 5]
```diff
@@ -4,21 +4,24 @@
...
 @app.route('/', methods=['GET', 'POST'])
 def index():
-    if request.form.get('page') == 'common':
+    if request.form.get('page') == 'turn_around_times':
+        year = request.form.get('year')
+        return redirect(url_for('turn_around_times', year_of_interest=year))
+    if request.form.get('page') == 'samples':
         year = request.form.get('year')
-        return redirect(url_for('common', year_of_interest=year))
+        return redirect(url_for('comon_samples', year_of_interest=year))
```
Comment (hassanfa) [inline vogue/server/views.py]: typo: common_samples

File: vogue/server/views.py [hunk 6]
```diff
@@ -31,118 +34,176 @@ def index():
...
         received_to_prepped = received_to_prepped,
         prepped_to_sequenced = prepped_to_sequenced,
         sequenced_to_delivered = sequenced_to_delivered,
         year_of_interest=year_of_interest,
-        this_year = THIS_YEAR,
         years = YEARS)
 
+
+@app.route('/common/samples/<year_of_interest>')
+def comon_samples(year_of_interest):
```
Comment (hassanfa) [inline vogue/server/views.py]: typo: common_samples

File: vogue/server/utils.py (hunk 1 above)
Comment (moonso) [reply vogue/server/utils.py]: It suggest `vogue/constants` where we can collect stuff like this

File: vogue/server/views.py [hunk 7]
```diff
@@ -4,21 +4,24 @@
...
-from vogue.server.utils import (find_recived_per_month, turn_around_times, 
-                                find_concentration_defrosts, find_concentration_amount,   
+from extentions import app
+from vogue.server.utils import ( find_concentration_defrosts, find_concentration_amount,   
                                 find_key_over_time)
 
 from  datetime import date
 
 THIS_YEAR = date.today().year
 YEARS = [str(y) for y in range(2017, THIS_YEAR + 1)]
```
Comment (moonso) [inline vogue/server/views.py]: Similar to the other constants this could be removed to `vogue/constants`

**Pull request merged**